storage/backend.go

```diff
@@ -57,6 +57,7 @@ type Backend interface {
 	GetTenants() []Tenant
 	GetItemList(tenant string, tags map[string]string) []Item
 	GetRawData(tenant string, id string, end int64, start int64, limit int64, order string) []DataItem
+	GetLastDataItems(tenant string, id string, numOfItems int) ([]DataItem, error)
```

Comment:
a. we need an `end` param to support aggregations  over T sec ago
e.g.  alert if `avg` over last 30 minuts was less then 300

b. we need to get all data points even if it was less the `limit` to support `count` aggregations.
e.g. alert if over the last 10 minutes we only got 2 new values from a server that suppose to give new value every 1 minutes.

c. a `start` will also be nice, so we can have compound alerts like:
alert 1 - yesterday the avg of a point was over X and today
alert 2 - today the avg of a point was over X and today
alert 3 - alert 1 or alert 2

we wan to support:

|aggregation | on        |    using current func     |    missing         |
|----------------|----------|------------------------------|---------------------|
|                    | last Item  |       `limit` = 1               |                         |
| sum,avg,min,max                   | last N Items  |       `limit` = N               |                         |
| sum,avg,min,max                   | last T sec  |      missing `end`             |    no `end` variable to set T minutes/sec ago                    |
| count                   | last T sec  |      missing `end`              |    we get nil instead of M last items, no `end` variable to set T minutes/sec ago                    |
| sum,avg,min,max,count                   | some past time span  |      missing `end` + `start`             |    we get nil instead of M last items, no `end` variable to set T minutes/sec ago                    |
| first item in group                   | N items or T sec  |      missing `order`             |    we only get last items                    |

Reply:
@yaacov I think in the beginning it's good just to support the top two types of alerts (which is what I had in mind) and then later add the other complex alerting. Trying to do everything at once is hard for me to do GOOD... I prefer to do the simple alerts and have them tested and be good and after that move forward to the other ones.

Reply:
Greate 👍 

Then for now just use 
GetRawData with:
start: 1 hour ago  
end: now
limit: 1
And we will refactor it in the future